kalman_watch/monitor_win_perigee.py

```diff
@@ -0,0 +1,768 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Perigee monitor window data and kalman drops
+
+Generate a trending plot which combines the perigee monitor window data during NMAN with
+the kalman drops data (from AOKALSTR) during NPNT. This is used to evaluate the
+evolution of the high ionizing radiation (IR) zone during perigee.
+
+This is normally run as a script which by default generates a plot
+``<data_dir>/mon_win_kalman_drops_-45d_-1d.png`` in the current directory. It will also
+cache the MAUDE images in the ``<data_dir>/aca_imgs_cache/`` directory.
+"""
+
+import argparse
+import functools
+import os
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TypeAlias
+
+os.environ["MPLBACKEND"] = "Agg"
```

Comment:
The old strategy relied on 
```
import matplotlib
matplotlib.use("Agg")
```
But in modern code that `import matplotlib` is sorted alphabetically and might land after another import that already imports `matlplotlib.pyplot`. This will activate the local interactive backend.

With this new strategy you can do the setup just after the system imports which are guaranteed not to import matplotlib. It also looks a bit tidier.

Reply:
I thought we had just been moving to putting matplotlib.use('Agg') into main().  My first concern is that using an environment variable seems to be a step back in that then importing has "side-effects"  in main) but I suppose for a trending application we're unlikely to use it from other code.

Reply:
Indeed this is not intended as an importable module, which is the key. If we wanted these functions to be importable then you are 100% right that we would need a separate application script that does the backend setup. 

The issue with backend setup in `main()` is that by the time it gets there all the imports have already happened, including (potentially) pyplot. I observed this in practice by seeing the matplotlib app sometimes appear and quickly disappear in my toolbar when running this script.